rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/CatalogRequestBuilder.java

```diff
@@ -60,57 +54,44 @@ public CatalogRequestBuilder(CatalogObjectAction catalogObjectAction) {
     }
 
     public String build() {
-        String urlEncodedCommitMessage = getUrlEncodedCommitMessageOrFail();
         StringBuilder builder = new StringBuilder(URL_CATALOG).append("/buckets/")
                                                               .append(this.catalogObjectAction.getBucketName());
         if (this.catalogObjectAction.isRevised()) {
             builder.append("/resources/").append(this.catalogObjectAction.getNodeSourceName()).append("/revisions?");
-            appendParameters(builder, paramPair(COMMIT_MESSAGE_PARAM, urlEncodedCommitMessage));
         } else {
             builder.append("/resources?");
-            appendParameters(builder,
-                             paramPair(NAME_PARAM, this.catalogObjectAction.getNodeSourceName()),
-                             paramPair(KIND_PARAM, this.catalogObjectAction.getKind()),
-                             paramPair(COMMIT_MESSAGE_PARAM, urlEncodedCommitMessage),
-                             paramPair(OBJECT_CONTENT_TYPE_PARAM, this.catalogObjectAction.getObjectContentType()));
         }
         return builder.toString();
     }
 
-    private String getUrlEncodedCommitMessageOrFail() {
-        try {
-            return URLEncoder.encode(this.catalogObjectAction.getCommitMessage(), "UTF-8");
-        } catch (UnsupportedEncodingException e) {
-            throw new IllegalArgumentException("Commit message cannot be encoded properly", e);
-        }
-    }
-
-    private void appendParameters(StringBuilder builder, String... paramPairs) {
-        builder.append(Arrays.stream(paramPairs).collect(Collectors.joining("&")));
-    }
-
-    public CloseableHttpResponse postNodeSourceRequestToCatalog(String sessionId, String fullUri)
+    public CloseableHttpResponse postNodeSourceRequestToCatalog(CatalogObjectAction catalogObjectAction, String fullUri)
             throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException, IOException {
-        HttpPost postNodeSource = buildCatalogRequest(sessionId, fullUri);
+        HttpPost postNodeSource = buildCatalogRequest(catalogObjectAction, fullUri);
         return getHttpClientBuilder().build().execute(postNodeSource);
     }
 
     private String paramPair(String param, String value) {
         return param + "=" + value;
     }
 
-    private HttpPost buildCatalogRequest(String sessionId, String fullUri) {
+    private HttpPost buildCatalogRequest(CatalogObjectAction catalogObjectActio, String fullUri) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "catalogObjectActio". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)